Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+                    strongSelf.resetPlayerFlags()
+                })
+                
+            case let e where e.self == PlayerEvent.ended:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("ended event: \(event)")
+                    
+                    strongSelf.sendPercentageReachedEvent(percentage: 100)
+                    strongSelf.reportView()
+                    strongSelf.stopViewTimer()
+                    strongSelf.resetPlayerFlags()
+                })
+                
+            case let e where e.self == PlayerEvent.playbackInfo:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("playbackInfo event: \(event)")
+                    strongSelf.kavaData.indicatedBitrate = event.playbackInfo?.indicatedBitrate
+                })
+            
+            case let e where e.self == PlayerEvent.trackChanged:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("trackChanged event: \(event)")
+                    if let track = event.selectedTrack {
+                        if (track.title.contains("event.selectedTrack?.title")) {
+                            strongSelf.kavaData.currentCaptionLanguage = event.selectedTrack?.language
+                            strongSelf.sendAnalyticsEvent(event: KavaEventType.captions)
+                        }
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.error:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("PlayerEvent error event: \(event)")
+                    strongSelf.kavaData.errorCode = (event.error?.code)!
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.error)
+                })
+            default: assertionFailure("all events must be handled")
+            }
+        }
+        
+        self.messageBus?.addObserver(self, events: [AdEvent.error], block: { [weak self] (event) in
+            guard let strongSelf = self else { return }
+            PKLog.debug("AdEvent error event: \(event)")
+            strongSelf.kavaData.errorCode = (event.error?.code)!
+            strongSelf.sendAnalyticsEvent(event: KavaEventType.error)
+        })
+    }
+    
+    /************************************************************/
+    // MARK: Private Implementation
+    /************************************************************/
+    
+    private func handleStateChanged(event: PKEvent) {
```

Comment:
If we only handle two states why do we need all the rest here? for logging this is also enabled on the state machine on playkit so this will just add verbose logs